webapp/app/forms/fields.py

```diff
@@ -19,6 +19,44 @@ def pre_validate(self, form):
         ValidElsterCharacterSet().__call__(form, self)
 
 
+class UnlockCodeWidget(TextInput):
+    """A divided input field with three text input fields, limited to four chars."""
+
+    def __call__(self, field, **kwargs):
+        if 'required' not in kwargs and 'required' in getattr(field, 'flags', []):
+            kwargs['required'] = True
+
+        kwargs['maxlength'] = 4
+        kwargs['class'] = 'form-control'
+
+        html = ""
+        kwargs['value'] = field._value()[0]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
+        html += '-'
+        kwargs['value'] = field._value()[1]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
+        html += '-'
+        kwargs['value'] = field._value()[2]
+        html += '<input %s>' % self.html_params(name=field.name, **kwargs)
+
+        return Markup(html)
+
+
+class UnlockCodeField(SteuerlotseStringField):
+    def __init__(self, label='', validators=None, **kwargs):
+        super(UnlockCodeField, self).__init__(label, validators, **kwargs)
+        self.widget = UnlockCodeWidget()
+
+    def process_formdata(self, valuelist):
+        if valuelist:
+            self.data = '-'.join(valuelist)
+        elif self.data is None:
+            self.data = ''
```

Comment:
We do this for the summary page and then use _value anywhere else? Or why are we joining and splitting the values?

Reply:
Not quite. _value is the value that gets displayed and we use self.data everywhere else - especially actually using the value - when sending to Elster for example.